File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.java
class Solution {

    public void checkPalindrome(String s, int l, int r, int[] resIdx, int[] resLen){
        while(l >= 0 && r < s.length() && s.charAt(l) == s.charAt(r)){
                if(r-l+1 > resLen[0]){
                    resLen[0] = r - l + 1;
                    resIdx[0] = l;
                }
                l--;
                r++;
            }
    }

    public String longestPalindrome(String s) {
        int[] resIdx = new int[1];
        int[] resLen = new int[1];
        for(int i=0; i < s.length(); i++){
            checkPalindrome(s, i, i, resIdx, resLen);
            checkPalindrome(s, i, i+1, resIdx, resLen);
        }
        return s.substring(resIdx[0], resIdx[0] + resLen[0]);
    }
}

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        resIdx, resLen = 0, 0
        for i in range(len(s)):
            resLen, resIdx = self.checkPalindrome(s, i, i, resIdx, resLen)
            resLen, resIdx = self.checkPalindrome(s, i, i + 1, resIdx, resLen)
        return s[resIdx:resIdx + resLen]

    def checkPalindrome(self, s, l, r , resIdx, resLen):
        while l >= 0 and r < len(s) and s[l] == s[r]:
                if resLen < r - l + 1:
                    resLen = r - l + 1
                    resIdx = l
                l-=1
                r+=1
        return resLen, resIdx